Sources/KSCrashRecording/KSCrashReportStoreC.c

```diff
@@ -61,38 +58,32 @@ static int compareInt64(const void *a, const void *b)
 
 static inline int64_t getNextUniqueID(void) { return g_nextUniqueIDHigh + g_nextUniqueIDLow++; }
 
-static void getCrashReportPathByID(int64_t id, char *pathBuffer)
+static void getCrashReportPathByID(int64_t id, char *pathBuffer, const KSCrashReportStoreCConfiguration *const config)
 {
-    assert(g_reportsPath != NULL);
-    snprintf(pathBuffer, KSCRS_MAX_PATH_LENGTH, "%s/%s-report-%016llx.json", g_reportsPath, g_appName, id);
+    snprintf(pathBuffer, KSCRS_MAX_PATH_LENGTH, "%s/%s-report-%016llx.json", config->reportsPath, config->appName, id);
 }
 
-static int64_t getReportIDFromFilename(const char *filename)
+static int64_t getReportIDFromFilename(const char *filename, const KSCrashReportStoreCConfiguration *const config)
 {
     char scanFormat[100];
-    sprintf(scanFormat, "%s-report-%%" PRIx64 ".json", g_appName);
+    sprintf(scanFormat, "%s-report-%%" PRIx64 ".json", config->appName);
 
     int64_t reportID = 0;
     sscanf(filename, scanFormat, &reportID);
     return reportID;
 }
 
-static int getReportCount(void)
+static int getReportCount(const KSCrashReportStoreCConfiguration *const config)
 {
-    if (g_reportsPath == NULL) {
-        KSLOG_ERROR("Reports store is not set up");
-        return 0;
-    }
-
     int count = 0;
-    DIR *dir = opendir(g_reportsPath);
+    DIR *dir = opendir(config->reportsPath);
```

Comment:
Is it possible to use `KSFileUtils` instead of the low-level API?

Reply:
There's no such API in the utils.